### PR TITLE
docs: repository-wide accuracy audit, with tests to keep it accurate

### PR DESCRIPTION
**Pull request by ginccc**
Cross-checked every claim in `docs/` and the root markdown against the source rather than against other documentation, fixed what was wrong, and added tests so the categories that rotted silently can't rot silently again.

The findings had one shape: **documentation rots wherever a wrong answer still produces a plausible response.** A renamed class breaks the build. A renamed property, metric, or REST path does not — `@ConfigProperty` resolves by string, Micrometer accepts any name, and `LegacyPathRewriteFilter` keeps pre-v6 paths answering.

## Could not work as written

| Doc | Problem |
|---|---|
| `incident-response.md` | A **breach runbook** where the log endpoint (`/admin/logs`) and **all three** named metrics were wrong — none of `eddi.conversations.active`, `eddi.tool.execution.count`, `eddi.audit.entries.count` is registered |
| `metrics.md`, `langchain.md` | Six tool endpoints under the pre-v6 `/langchain/tools`. They *answered*, because `LegacyPathRewriteFilter` rewrites them — which is why nobody noticed. `GET /llm/toolhistory/costs` is a plain 404 |
| `conversations.md` | Documented a `redoCacheSize` field, with three bullets interpreting its values. It has never existed — the DTO carries `undoAvailable`/`redoAvailable` |
| `hipaa-compliance.md` | Told operators to set `eddi.usermemory.auto-purge-days`. No such property; the real one ships as `-1`, i.e. off |
| `attachments-guide.md`, `architecture.md` | Described `MultimodalMessageEnhancer`, deleted in 6.1.0. PDF and audio listed as "future" when both ship |
| `behavior-rules.md` | v5 `BehaviorSet` model, and `/currentversion` given a ruleset body when `GET` returns a bare integer and `POST` takes none |
| `architecture.md` | `POST /agentstore/{id}/signing/keys` does not exist |
| `security.md`, `architecture.md` | Stale `modules.langchain.tools` package paths (now `modules.llm.tools`) |
| `README.md`, `AGENTS.md` | Both said `./mvnw verify` runs integration tests. `skipITs=true` — it doesn't. CI passes `-DskipITs=false` |
| `getting-started.md` | Compose v1 syntax, the v5 word "packages", a k8s step running a repo script with no checkout, and **no success signal at all** |

## A claim with nothing behind it

`metrics.md` promised the Full Metrics dashboard "covers all 144 registered meters — it is generated from the registration sites in the source, so a metric cannot be added to the codebase and silently go unwatched."

Nothing generated it and nothing checked it, and it was **already false**: five `eddi.llm.cascade.*` counters were registered and on no panel — precisely the meters that say whether cascading saves money or pays twice per turn. Five panels added; the claim replaced with one that is enforced.

## Added

- **`docs/configuration-reference.md`** — all 118 `eddi.*` properties with default, env-var spelling and behaviour. **61 were documented nowhere**, including `eddi.security.ssrf-protection.enabled` (off by default), and every `eddi.schedule.*`, `eddi.shutdown.*` and `eddi.nats.*`.
- **`scheduling.md`** — deployment configuration and metrics. The page explained cluster-awareness without mentioning `lease-timeout` or `instance-id`, and never said delivery is at-least-once, so targets must be idempotent.
- **`security.md`** — the SSRF section described unconditional protection. It is unconditional for *tool* URLs, but httpCall/MCP/A2A targets sit behind an opt-in flag that defaulted to `false` and appeared in no document.
- **`metrics.md`** — 66 registered meters were missing. Thirteen new sections; coverage now 135/135.

## Changelog: 1.9 MB → 184 KB

`docs/changelog.md` had reached **1.9 MB across 561 entries** (~500k tokens) and was linked from `SUMMARY.md` as a browsable page. AGENTS.md rule 8 obliges every session to append and never to remove, so the growth was structural.

Split into per-month archives under `docs/changelog/`. Total bytes unchanged (1.89 MB both sides); reconciliation shows **0 missing headings, 0 lost content lines**.

- **Monthly, not per-release** — entries carry dates, not release tags.
- **Size-triggered rotation, not calendar-triggered** — a rule that fires on the 1st is one somebody has to remember. `ChangelogRotationTest` fails past 250 KB, and its message says *rotate, don't raise the cap*.
- **`Decision Log` and `Regression Notes` stay live** — they are running registers, not entries; archiving them would retire both silently.
- **`scripts/rotate-changelog.py`** makes the procedure executable. The mechanical part is what a hand-rotation gets wrong: an earlier pass of this same logic "fixed" the `alt` rows *inside code spans* in two entries — documentation of link syntax, not links.
- **The cap is measured on CRLF-normalised content.** `Files.size()` would have made it ~5% tighter on Windows than on Linux CI, for byte-identical content.

## Tests, so this does not recur

Link rot and legacy paths already had guards. Configuration and metrics had none — which is where the rot was.

- `MetricsDashboardCoverageTest` — every registered meter needs a dashboard panel **and** a `metrics.md` entry.
- `ConfigurationReferenceCoverageTest` — the reference must be exhaustive **and** invent nothing. The second direction matters as much: a documented property nothing reads is a silent no-op, and the operator believes the deployment is configured when it is not.
- `ChangelogRotationTest` — the cap, archive naming/indexing, and the survival of the two registers.
- `DocumentedRestPathsTest` — `/langchain/tools` and `/bottriggerstore/bottriggers` added; the exemption list made prefix-aware so a routine rotation doesn't fail it.

**All new assertions were mutation-checked** — each confirmed to fail when the fix it guards is reverted.

## Moved

`HANDOFF.md` → `docs/archive/handoff-v6.0-snapshot.md`. 70 KB, last updated 2026-03-30, self-declared "no longer actively maintained", referenced by nothing, sitting at the repository root where coding assistants load it — full of renamed classes and pre-v6 paths. It was already exempted from `DocumentedRestPathsTest` for exactly that reason.

---

No production code is touched: the diff is documentation, three new test classes, one modified test class, and one script.

## Summary by CodeRabbit

* **Documentation**
  * Added a comprehensive configuration reference covering properties, environment variables, defaults, and deployment settings.
  * Updated setup, security, scheduling, monitoring, attachment, conversation, and API guidance.
  * Clarified build commands, integration-test behavior, versioned rulesets, current REST paths, and Kubernetes deployment limits.
  * Added monthly changelog archives and improved maintenance guidance.
* **New Features**
  * Added automated changelog rotation with size-cap checking.
  * Expanded dashboard coverage with model cascade and streaming metrics.
* **Bug Fixes**
  * Corrected outdated component names, endpoint paths, configuration properties, and architecture references.

### CODE REVIEW

File: src/test/java/ai/labs/eddi/docs/ChangelogRotationTest.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        var problems = new TreeSet<String>();
+
+        for (Path file : listMarkdown(archives)) {
+            String name = file.getFileName().toString();
+            Matcher m = ARCHIVE_NAME.matcher(name);
+            if (!m.matches()) {
+                problems.add(name + " — archives must be named YYYY-MM.md, one per month");
+                continue;
+            }
+            int month = Integer.parseInt(m.group(2));
```
[the remark this answers is not in the document]
Comment (ginccc) [reply src/test/java/ai/labs/eddi/docs/ChangelogRotationTest.java]: Fixed in 10537154b — though by removing the parse rather than guarding it.

You are right that a reader has to reason to dismiss this, and that is reason enough to change it. `NumberFormatException` could not actually be thrown: `group(2)` came from `(\d{2})`, so two digits were already established by the time `parseInt` ran. But a validation whose safety depends on reading a regex three lines away is worse than one that cannot fail at all.

The pattern now says "a real month" declaratively — `^(\d{4})-(0[1-9]|1[0-2])\.md$` — and the `parseInt` and range check are gone. Same validation, one step, nothing to catch.

File: docs/attachments-guide.md [hunk 2]
```diff
@@ -77,7 +77,7 @@ Client sends context with attachment_* keys
 1. **Context Extraction** — `AttachmentContextExtractor` parses `attachment_*` context keys into `Attachment` objects
 2. **Memory Storage** — Attachments are stored as `List<Attachment>` in the `attachments` memory key
 3. **Rule Matching** — `ContentTypeMatcher` condition matches on MIME types for routing
-4. **LLM Forwarding** — `MultimodalMessageEnhancer` converts attachments to langchain4j `ImageContent` and enhances the user message
+4. **LLM Forwarding** — `AttachmentForwarder` resolves each attachment's bytes, gates it on `ModelCapabilityService`, and converts it to the right langchain4j `Content` on the outgoing user message
```
[the remark this answers is not in the document]
Comment (ginccc) [reply docs/attachments-guide.md:89]: Good catch, and thank you — this is one my own sweep could not have found.

The diagram spells the name across four lines as `Multi-` / `modal` / `Message` / `Enhancer`, so the string search that cleaned up every other `MultimodalMessageEnhancer` reference in the prose never matched it. A stale identifier hiding from grep by being word-wrapped is exactly the kind of thing this PR is about.

Fixed in 10537154b. The box now describes `AttachmentForwarder` — resolving bytes, gating on `ModelCapabilityService`, emitting langchain4j `Content` — matching stage 4 below it.

While in there I also fixed the geometry: the top box had a 36-character border around 41-character content, and the three lower boxes each had a different width from their own borders. All boxes are now internally consistent and the connectors are centred on them.

File: docs/configuration-reference.md [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+```
+eddi.schedule.poll-interval        →  EDDI_SCHEDULE_POLLINTERVAL
+eddi.vault.master-key              →  EDDI_VAULT_MASTERKEY
+eddi.openai-compat.api-key         →  EDDI_OPENAICOMPAT_APIKEY
+eddi.tools.websearch.google.api-key →  EDDI_TOOLS_WEBSEARCH_GOOGLE_APIKEY
+```
+
+`EDDI_VAULT_MASTER_KEY` (with the extra underscore) also works, because
+MicroProfile tries several transformations — but `EDDI_SCHEDULE_POLL_INTERVAL`
+is *not* guaranteed to, so prefer the mechanical form above and you will never
```
[the remark this answers is not in the document]
Comment (ginccc) [reply docs/configuration-reference.md]: You are right, and this was the most serious defect in the PR. Fixed in 10537154b.

I claimed MicroProfile Config "deletes `-` entirely". It does not — **both `.` and `-` are replaced with `_`**. The repository had been telling me so the whole time: `EDDI_VAULT_MASTER_KEY`, `EDDI_MCP_ALLOW_UNAUTHENTICATED` and `EDDI_OPENAI_COMPAT_API_KEY` are the spellings already in `docker-compose.yml`, `k8s/` and `AuditHmac`'s own javadoc. I wrote the rule from memory and never checked the reference against the deployment assets sitting next to it.

Your Security tag is the right severity. An unrecognised environment variable **is not an error** — the property keeps its default and the service starts normally — so `EDDI_VAULT_MASTERKEY` leaves `eddi.vault.master-key` empty, the vault inactive, and `scope: "secret"` properties falling back to plaintext, with nothing in the startup log naming the variable that was set. The section now leads with that consequence instead of the rule, and `EDDI_VAULT_MASTERKEY` is kept only as an explicitly labelled counter-example.

Both sites are corrected (`configuration-reference.md` and the `scheduling.md` example).

Because a wrong rule can be reintroduced as easily as it was written, `ConfigurationReferenceCoverageTest` now asserts that every `EDDI_*` token in the documentation is the mechanical transform of a real property. It immediately found two further instances neither of us had flagged: `EDDI_VERSION` (a Compose image tag, not a property) and `EDDI_AUDIT_RETENTIONDAYS` (deliberately named in the note explaining that property's removal). Both are now classified explicitly rather than left ambiguous.

The general lesson, which I have written into the changelog: a reference checked for *coverage* is not thereby checked for *correctness*. My existing test verified that all 118 properties were present; it had nothing to say about whether the page described them accurately.

File: docs/metrics.md [hunk 4]
```diff
@@ -286,6 +290,179 @@ sum(rate(eddi_tenant_quota_denied_total[5m]))
 gates (`checkAgentQuota`, `checkCostBudget`) deliberately do not touch it, so
 `allowed / (allowed + denied)` is not a true accept rate.
 
+### Coordinator Metrics
+
+```
```
[the remark this answers is not in the document]
Comment (ginccc) [reply docs/metrics.md]: Fixed in 10537154b, and applied more widely than reported.

MD040 flagged the 13 fences this PR added, but `metrics.md` already had 29 bare fences against 12 tagged ones — the new sections were following the file's existing style. Tagging only mine would have left the file half-converted and the next contributor guessing which convention to copy.

All 29 now carry `text`, which is the identifier markdownlint's own documentation recommends for content with no syntax to highlight. No bare fences remain in the file.

File: src/test/java/ai/labs/eddi/docs/ChangelogRotationTest.java [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+            if (month < 1 || month > 12) {
+                problems.add(name + " — month " + m.group(2) + " is not a month");
+            }
+            // The Archive table is the only way a reader finds these; an
+            // unlisted file is rotated-away-and-lost rather than archived.
+            if (!live.contains("changelog/" + name)) {
+                problems.add(name + " — not linked from the Archive table in docs/changelog.md");
+            }
```
[the remark this answers is not in the document]
Comment (ginccc) [reply src/test/java/ai/labs/eddi/docs/ChangelogRotationTest.java]: Confirmed and fixed in 10537154b. This is a real weakness, and an uncomfortable one given the test exists specifically to catch index drift.

`live.contains("changelog/" + name)` matched anywhere in the file. That is not hypothetical here: this changelog now contains entries *about* the rotation mechanism that discuss `docs/changelog/` paths, so the check was roughly one edit away from passing vacuously on precisely the drift it guards.

The match is now scoped to a Markdown table row within the `## Archive` section:

```java
Pattern.compile("^\|.*\]\(changelog/" + Pattern.quote(fileName) + "\).*\|\s*$", Pattern.MULTILINE)
```

against a substring extracted from `## Archive` up to the next `##` heading, with an assertion that the section exists at all — a missing index is itself a failure, not a reason to skip the check.

Mutation-checked: removing the `2026-05.md` table row while leaving a prose mention of `changelog/2026-05.md` elsewhere in the file now fails, where before it passed.

File: src/test/java/ai/labs/eddi/docs/ChangelogRotationTest.java [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+        for (Path file : listMarkdown(archives)) {
+            String name = file.getFileName().toString();
+            if (!ARCHIVE_NAME.matcher(name).matches()) {
+                problems.add(name + " — archives must be named YYYY-MM.md, with a real month");
+                continue;
+            }
+            // The Archive table is the only way a reader finds these; a file
+            // missing from it is rotated-away-and-lost rather than archived.
+            if (!archiveTableRow(name).matcher(archiveTable).find()) {
+                problems.add(name + " — no row for it in the ## Archive table of docs/changelog.md");
```
[the remark this answers is not in the document]
Comment (ginccc) [reply src/test/java/ai/labs/eddi/docs/ChangelogRotationTest.java:134]: Confirmed and fixed in a3c0a4afc. Both directions are now checked.

I verified the failure you describe before changing anything: deleting `docs/changelog/2026-05.md` while leaving its table row made `ChangelogRotationTest` pass 3/3. So the test that owns this invariant was blind to half of it.

One nuance worth recording — the resulting broken link *was* caught, by `DocumentationLinksTest`, which fails on any unresolved relative target. But that is the wrong test to hear it from: it reports a generic unresolved link and says nothing about the Archive index being stale, so the reader has to work backwards to the cause. And it cannot catch your second point at all — a row naming a file that exists but under a name no rotation would ever produce, e.g. `changelog/notes.md`.

So the reverse pass applies `ARCHIVE_NAME` to indexed names as well as to files on disk, and reports the two cases distinctly:

- `notes.md — indexed in the ## Archive table, but not a YYYY-MM.md archive name`
- `2026-05.md — a row in the ## Archive table points at it, but the file does not exist`

Both mutation-checked. The first now fails where it previously passed; the second now fails *here* rather than only in `DocumentationLinksTest`.

File: src/test/java/ai/labs/eddi/docs/ConfigurationReferenceCoverageTest.java [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+        for (String property : collectProperties(root).keySet()) {
+            valid.add(property.replaceAll("[^A-Za-z0-9]", "_").toUpperCase(Locale.ROOT));
+        }
+
+        var wrong = new TreeSet<String>();
+        for (Path file : documentationFiles(root)) {
+            String relative = root.relativize(file).toString().replace('\\', '/');
+            if (relative.startsWith("docs/changelog")) {
+                continue; // historical entries quote whatever was true at the time
+            }
```
[the remark this answers is not in the document]
Comment (ginccc) [reply src/test/java/ai/labs/eddi/docs/ConfigurationReferenceCoverageTest.java:151]: Both correct. Fixed in a3c0a4afc.

**The prefix was too broad.** `startsWith("docs/changelog")` would have exempted a `docs/changelog-notes.md` as well as the changelog itself. Now an exact match on `docs/changelog.md` plus a prefix on `docs/changelog/`. This is the failure mode that actually matters for an exemption: exempting *more* than intended is invisible, because the test goes on passing.

**The file list contradicted the PR description.** I wrote that the audit covers `docs/` and root markdown, then enumerated exactly two root files. `PRIVACY.md`, `CONTRIBUTING.md`, `SECURITY.md`, `GOVERNANCE.md` and `CODE_OF_CONDUCT.md` were unread — and `PRIVACY.md` is a 30 KB operator-facing document, which is precisely where a configuration name gets quoted.

Every root `*.md` is now enumerated rather than allow-listed. I checked: none of the newly included files names an `EDDI_*` variable today, which is exactly why the allow-list would have gone on looking correct indefinitely — there was no failing case to reveal it.

Mutation-checked by appending `EDDI_VAULT_MASTERKEYY=x` to `PRIVACY.md`; the test now reports `PRIVACY.md:473 EDDI_VAULT_MASTERKEYY`, where before that file was never opened.

File: docs/compliance-data-flow.md [hunk 8]
```diff
@@ -125,15 +131,25 @@ User Input (may contain PII)
...
+12. Audit Ledger ─────────────── userId → SHA-256 PSEUDONYMIZED
+    Audit Ledger Event ───────── GDPR_ERASURE entry written (immutable)
 ```
 
-Steps 4–5 retain operational and compliance data but make re-identification
+Steps 2–5 run before the conversation snapshots are deleted because they
+reference conversation IDs that the bulk delete removes.
+
+Steps 11–12 retain operational and compliance data but make re-identification
 impossible without the original userId.
```
[the remark this answers is not in the document]
Comment (ginccc) [reply docs/compliance-data-flow.md]: Confirmed and fixed in 9cfc367c0. This was the most consequential finding on the PR.

`AuditHmac.pseudonymFor` is `GDPR_PSEUDONYM_PREFIX + sha256Hex(userId)` — **unsalted and unkeyed**. It is deterministic, so an observer holding a list of candidate user IDs hashes each one and matches. "Impossible" was wrong on the technical facts.

It was worse than a technical inaccuracy, though. "Re-identification impossible" describes *anonymisation*, and anonymised data falls outside GDPR entirely — so the sentence invited an operator to disclose audit records on a basis the implementation does not support. The page now says:

> This is pseudonymisation, not anonymisation, and the distinction is legal as well as technical. […] Under GDPR Art. 4(5) pseudonymised data remains **personal data** and stays in scope. Do not treat these records as anonymised, and do not disclose them on that basis.

I deliberately did **not** propose salting or keying the digest here. The pseudonym has to stay deterministic — `AuditHmac.identityToken` relies on `pseudonymFor` being reproducible so that replacing an identifier with its own pseudonym is the one substitution that still verifies against the HMAC chain. Changing the derivation would break audit verification for every previously-erased user. If the unsalted digest is judged insufficient, that is a design change with a migration, not a docs edit.

File: docs/docker.md [hunk 9]
```diff
@@ -42,9 +42,17 @@ docker run --name mongodb -d mongo:6.0
...
+docker run --name eddi \
+  --link mongodb:mongodb \
+  -p 7070:7070 \
+  -e EDDI_SECURITY_ALLOW_UNAUTHENTICATED=true \
+  -e EDDI_MCP_ALLOW_UNAUTHENTICATED=true \
+  -e EDDI_SECRETSTORE_ALLOW_UNAUTHENTICATED=true \
+  -d labsai/eddi:latest
 ```
 
+> **Note:** The image runs in production mode, where `AuthStartupGuard` and `HighValueSurfaceGuard` refuse to boot while OIDC is off. Without these three opt-outs the container exits at startup. This is exactly what `docker-compose.yml` sets for you.
```
[the remark this answers is not in the document]
Comment (ginccc) [reply docs/docker.md:60]: Confirmed and fixed in 9cfc367c0.

You are right that the exposure is the point, not the flags. The three opt-outs disable authentication on *every* endpoint — `/secretstore` is the vault and `/mcp` is agent CRUD — and the example published `7070` on all interfaces.

Changed the publish to `-p 127.0.0.1:7070:7070` so the container is not reachable off-host, retitled the section "**local development only**", and added a warning that names what is exposed and points at the authenticated example below it.

I kept the existing note about `AuthStartupGuard`/`HighValueSurfaceGuard`, since it explains *why* the flags are needed at all — but it now sits under the warning rather than standing in for one.

File: docs/kubernetes.md [hunk 10]
```diff
@@ -21,11 +21,16 @@ kubectl apply -f https://raw.githubusercontent.com/labsai/EDDI/main/k8s/quicksta
...
+# "application-secrets.properties".
+kubectl delete secret eddi-secrets -n eddi --ignore-not-found
+printf 'eddi.vault.master-key=%s\n' "$(openssl rand -base64 24)" \
+  > /tmp/application-secrets.properties
 kubectl create secret generic eddi-secrets \
   --namespace=eddi \
-  --from-literal=EDDI_VAULT_MASTER_KEY="$(openssl rand -base64 24)" \
-  --dry-run=client -o yaml | kubectl apply -f -
+  --from-file=/tmp/application-secrets.properties
+shred -u /tmp/application-secrets.properties
```
[the remark this answers is not in the document]
Comment (ginccc) [reply docs/kubernetes.md]: Confirmed and fixed in 9cfc367c0 — and this one is mine: that snippet was added earlier in this same branch, so the review caught a defect this work introduced rather than an inherited one.

Both halves were real. `printf > /tmp/application-secrets.properties` creates the file at the default umask (world-readable on most images) while it holds the vault master key, and if `kubectl create secret` failed the `shred` never ran, leaving it there.

Now:

```bash
( umask 077
  trap 'shred -u /tmp/application-secrets.properties 2>/dev/null' EXIT
  printf 'eddi.vault.master-key=%s\n' "$(openssl rand -base64 24)" \
    > /tmp/application-secrets.properties
  kubectl create secret generic eddi-secrets \
    --namespace=eddi \
    --from-file=/tmp/application-secrets.properties )
```

The subshell scopes the `umask` so it does not leak into the reader's session, and the `EXIT` trap fires on the failure path as well as the success path.

File: docs/mcp-client.md [hunk 11]
```diff
@@ -171,16 +171,17 @@ server holds per-user state, that matters; if it is stateless, it does not.
...
 
-`apiKey` is sent as `Authorization: Bearer <value>`. Three forms are resolved:
+`apiKey` is sent as `Authorization: Bearer <value>`. Four forms are resolved:
 
 | Value | Resolves to |
 | --- | --- |
 | `${vault:name}` | A vault secret. **Use this.** |
 | `${vars:name}` | A global variable — for non-secret values. |
 | `${caller:token}` | The chatting user's own bearer token, released only to the **same origin** the caller addressed. |
+| `${connection:name}` | A managed connection — OAuth or static, `SERVICE` or `PER_USER`; resolved per request. See [connections](connections.md). Must be the whole value (`ConnectionReference.requireSole`). |
```
[the remark this answers is not in the document]
Comment (ginccc) [reply docs/mcp-client.md:181]: Confirmed as a real gap, and documented in 9cfc367c0 — but deliberately **not** fixed in code here.

Verified: `McpToolProviderManager` validates only that the scheme is one of `http`/`https` (`MCP server URL must use http or https`), and nothing downstream refuses to attach a bearer to a cleartext connection. So an `apiKey` against an `http://` server goes out in the clear with no warning.

The page now says so:

> **Use `https://` for any server you send a credential to.** The URL validator accepts `http://` and `https://` alike, and nothing refuses to attach a bearer token to a cleartext connection […] Reserve `http://` for a loopback server with no credential.

I have not implemented the rejection you describe. This is a documentation branch, and refusing `http://` when `apiKey` is set is a **behavioural change that would break existing working configurations** — anyone pointing at an internal plaintext MCP server would fail at startup after upgrading. That needs its own change with a decision about whether it errors or warns, whether loopback is exempt, and a release note. Raising it as a follow-up rather than smuggling it in behind a docs review.

File: docs/redhat-openshift.md [hunk 12]
```diff
@@ -159,20 +159,29 @@ The operator creates a route automatically. With the CR above, the route would b
...
 For production deployments with MongoDB:
 
 ```bash
 docker run -d \
   -p 7070:7070 \
-  -e QUARKUS_MONGODB_CONNECTION_STRING=mongodb://mongo:27017 \
+  -e MONGODB_CONNECTIONSTRING='mongodb://mongo:27017/eddi?retryWrites=true&w=majority' \
+  -e EDDI_SECURITY_ALLOW_UNAUTHENTICATED=true \
+  -e EDDI_MCP_ALLOW_UNAUTHENTICATED=true \
+  -e EDDI_SECRETSTORE_ALLOW_UNAUTHENTICATED=true \
```
[the remark this answers is not in the document]
Comment (ginccc) [reply docs/redhat-openshift.md]: Confirmed and fixed in 9cfc367c0.

The heading said "For production deployments with MongoDB" and the command underneath set all three unauthenticated opt-outs while publishing `7070`. Those flags exist so a local container can boot past `AuthStartupGuard`; putting them in the block labelled *production* is precisely backwards.

The production example now configures OIDC instead:

```bash
-e QUARKUS_OIDC_TENANT_ENABLED=true \
-e QUARKUS_OIDC_AUTH_SERVER_URL='https://keycloak.example.com/realms/eddi' \
-e QUARKUS_OIDC_CLIENT_ID=eddi \
-e QUARKUS_OIDC_CREDENTIALS_SECRET="$OIDC_CLIENT_SECRET"
```

with a sentence above it saying what the opt-outs are for and what they leave open, so nobody reintroduces them by pattern-matching from the development example earlier on the page.

File: docs/kubernetes.md [hunk 13]
```diff
@@ -21,11 +21,19 @@ kubectl apply -f https://raw.githubusercontent.com/labsai/EDDI/main/k8s/quicksta
...
+kubectl delete secret eddi-secrets -n eddi --ignore-not-found
+# umask first: the file holds the master key, and the default mode is
+# world-readable on most images. The trap removes it even if kubectl fails.
+( umask 077
+  trap 'shred -u /tmp/application-secrets.properties 2>/dev/null' EXIT
+  printf 'eddi.vault.master-key=%s\n' "$(openssl rand -base64 24)" \
+    > /tmp/application-secrets.properties
+  kubectl create secret generic eddi-secrets \
+    --namespace=eddi \
+    --from-file=/tmp/application-secrets.properties )
```
[the remark this answers is not in the document]
Comment (ginccc) [reply docs/kubernetes.md]: Confirmed and fixed in 17552f729. You are right that `umask 077` was the wrong tool: it governs the mode of a file *you* create, and does nothing about another local user having already created that path — or replaced it with a symlink — before the redirection runs.

Now `mktemp`, which returns an unpredictable name already at `0600`:

```bash
secrets_file=$(mktemp "${TMPDIR:-/tmp}/eddi-secrets.XXXXXX")
trap 'shred -u "$secrets_file" 2>/dev/null || rm -f "$secrets_file"' EXIT
```

One correction to the suggested form, worth recording because it is a silent failure: **passing the generated path to both commands does not work as written.** `--from-file=<path>` names the Secret key after the file's basename, so an `mktemp` path yields a key called `eddi-secrets.a1B2c3`. The Deployment mounts exactly `application-secrets.properties` (`k8s/base/eddi-deployment.yaml`), so the pod would come up with no key file and the vault silently inert. The explicit form is required:

```bash
--from-file=application-secrets.properties="$secrets_file"
```

That is what `k8s/create-secrets.sh` has always done; these copies had drifted from it. Fixed in all four places the recipe appears — `docs/kubernetes.md` (twice), `docs/getting-started.md`, `k8s/base/eddi-secret.yaml`, `k8s/quickstart.yaml` — so the manifests and the docs now agree with the script. All three shell blocks pass `bash -n` and both manifests still parse.

[the remark this answers is not in the document]
Comment (ginccc) [reply docs/kubernetes.md]: Confirmed and fixed in 17552f729.

This was the more dangerous of the two, because the failure is invisible at every layer: `openssl rand` fails, `printf` still writes `eddi.vault.master-key=`, `kubectl create secret` succeeds, the pod starts healthy, and the vault is inert — so `scope: "secret"` properties fall back to plaintext with nothing in the startup log naming the cause. An operator would have no signal until they went looking.

Now fails closed before anything is written:

```bash
key=$(openssl rand -base64 24)
[ -n "$key" ] || { echo "vault key generation failed" >&2; exit 1; }
printf 'eddi.vault.master-key=%s\n' "$key" > "$secrets_file"
```

`set -euo pipefail` at the top of the block covers the non-empty-but-failed case, and the explicit `-n` test covers a zero-status empty result. Applied to all four copies of the recipe.

**Review by rolandpickl — APPROVED** [review]
